tests/jest/morph/internal.spec.js

```diff
@@ -1,50 +1,50 @@
 let { morph } = require('@alpinejs/morph')
 let createElement = require('./createElement.js')
 
-test('changed element is the same element', () => {
+test('changed element is the same element', async () => {
     let dom = createElement('<div><span>foo</span></div>')
 
     dom.querySelector('span').is_me = true
 
-    morph(dom, '<div><span>bar</span></div>')
+    await morph(dom, '<div><span>bar</span></div>')
 
     expect(dom.querySelector('span').is_me).toBeTruthy()
 })
 
-test('non-keyed elements are replaced instead of moved', () => {
+test('non-keyed elements are replaced instead of moved', async () => {
```

Comment:
TODO:

- Figure out the intent
- Figure out what went wrong where

Reply:
The intent is to demostrate that the first list item does not move. The test sets a custom property on the li tag then it triggers a mutation where the text content of the first list item changes and a second list item is added. Note, the new dom element looks like the first one before the mutation but it's actually a brand new node as the custom property is still set on the first item. If the node was keyed, the mutation would move the first node to position due and prepend a new node so the custom property would be on the second list item.

Reply:
Oh no I did get what the code was doing, my question was more like "_Why is this the expected behavior in the first place_". But I admit my phrasing wasn't the best, and thanks for the thorough answer :v:.

Now I figured this behavior is wanted when the `lookahead` option is set to `false`, otherwise the expected behavior is the opposite, meaning : "_non-keyed elements are moved instead of replaced_".

So my question now would go to @calebporzio directly because I'm confused as to what the default boolean value for the `lookahead` option should be.

The [docs](https://github.com/alpinejs/alpine/blob/main/packages/docs/src/en/plugins/morph.md?plain=1#L199) sais it should be `false` but it was changed to `true` in [#d07aeee](https://github.com/alpinejs/alpine/commit/d07aeee#diff-16dbe79737c2ee37a02e810f6102eac94023aa7204051573a1463b3a846f5824R60).

Reply:
Yep, it's already reverted to false on the main branch. 👍